src/java.base/share/native/libjli/java.c

```diff
@@ -1921,14 +1921,19 @@ DescribeModule(JNIEnv *env, char *optString)
  * Prints default usage or the Xusage message, see sun.launcher.LauncherHelper.java
  */
 static void
-PrintUsage(JNIEnv* env, jboolean doXUsage)
+PrintUsage(JNIEnv* env, jboolean doXUsage, jboolean conciseUsage)
 {
-  jmethodID initHelp, vmSelect, vmSynonym, printHelp, printXUsageMessage;
+  jmethodID initHelp, vmSelect, vmSynonym;
+  jmethodID printHelp, printConciseUsageMessage, printXUsageMessage;
   jstring jprogname, vm1, vm2;
   int i;
   jclass cls = GetLauncherHelperClass(env);
   NULL_CHECK(cls);
-  if (doXUsage) {
+  if (conciseUsage) {
+    NULL_CHECK(printConciseUsageMessage = (*env)->GetStaticMethodID(env, cls,
+                                        "printConciseUsageMessage", "(Z)V"));
+    (*env)->CallStaticVoidMethod(env, cls, printConciseUsageMessage, printTo);
```

Comment:
Should we have a exception check here after the `CallStaticVoidMethod` returns? I realize that several places in the current launcher code, in this file, don't have those checks, but the JNI specific of `CallStaticVoidMethod` does note that this function could raise any exception thrown by the underlying Java method?

Reply:
I see `CHECK_EXCEPTION_LEAVE(1);` in the caller of this method. Given we could only do `CHECK_EXCEPTION_RETURN();`, which returns if an exception is thrown (so that the caller can handle the exception as they see fit), it seems unnecessary for last calls. As for last calls, it is "return if an exception is thrown, and return if it is not". Note it is not only this function, `PrintJavaVersion`, `ShowSettings`, `ShowResolvedModules`, `ListModules` and maybe other use the same pattern, where the caller handles the exceptions, and the functions don't check the exceptions for the last call.

Reply:
That's a good point about this being a last call before the control returns out of this function. I had missed that part. What you note about not needing an exception check here, makes sense.